plugins/datetime/msd-datetime-mechanism.c

```diff
@@ -113,331 +163,337 @@ msd_datetime_mechanism_error_get_type (void)
+                                      "Error reading /etc/adjtime file: %s", error->message);
+                g_error_free (error);
+                g_dbus_method_invocation_return_gerror (invocation, error2);
+                g_error_free (error2);
+                return FALSE;
+        }
 
-        result = polkit_authority_check_authorization_sync (mechanism->priv->auth,
-                                                            subject,
-                                                            action,
-                                                            NULL,
-                                                            POLKIT_CHECK_AUTHORIZATION_FLAGS_ALLOW_USER_INTERACTION,
-                                                            NULL, &error);
-        g_object_unref (subject);
+        lines = g_strsplit (data, "\n", 0);
+        g_free (data);
 
-        if (error) {
-                dbus_g_method_return_error (context, error);
+        if (g_strv_length (lines) < 3) {
+                error = g_error_new (MSD_DATETIME_MECHANISM_ERROR,
+                                     MSD_DATETIME_MECHANISM_ERROR_GENERAL,
+                                     "Cannot parse /etc/adjtime");
+                g_dbus_method_invocation_return_gerror (invocation, error);
                 g_error_free (error);
-
+                g_strfreev (lines);
                 return FALSE;
         }
 
-        if (!polkit_authorization_result_get_is_authorized (result)) {
+        if (strcmp (lines[2], "UTC") == 0) {
+                is_utc = TRUE;
+        } else if (strcmp (lines[2], "LOCAL") == 0) {
+                is_utc = FALSE;
+        } else {
                 error = g_error_new (MSD_DATETIME_MECHANISM_ERROR,
-                                     MSD_DATETIME_MECHANISM_ERROR_NOT_PRIVILEGED,
-                                     "Not Authorized for action %s", action);
-                dbus_g_method_return_error (context, error);
+                                     MSD_DATETIME_MECHANISM_ERROR_GENERAL,
+                                     "Expected UTC or LOCAL at line 3 of /etc/adjtime; found '%s'",
+                                     lines[2]);
+                g_dbus_method_invocation_return_gerror (invocation, error);
                 g_error_free (error);
-                g_object_unref (result);
-
+                g_strfreev (lines);
                 return FALSE;
         }
+        g_strfreev (lines);
+        mate_settings_date_time_mechanism_complete_get_hardware_clock_using_utc (object, invocation, is_utc);
+        return TRUE;
+}
 
-        g_object_unref (result);
+static gboolean
+msd_datetime_mechanism_get_timezone_handler (MateSettingsDateTimeMechanism *object,
+                                             GDBusMethodInvocation         *invocation,
+                                             gpointer                       user_data G_GNUC_UNUSED)
+{
+        gchar *timezone;
```

Comment:
fond a buils warning in CI. https://app.travis-ci.com/github/mate-desktop/mate-settings-daemon/jobs/610161263#L4685
```
msd-datetime-mechanism.c:366:16: warning: declaration shadows a variable in the global scope [-Wshadow]

        gchar *timezone;

               ^

/usr/include/time.h:233:17: note: previous declaration is here

extern long int timezone;

                ^

1 warning generated.
```